datalake/src/main/java/com/sequenceiq/datalake/controller/sdx/SdxController.java

```diff
@@ -502,6 +503,22 @@ public void submitDatalakeDataSizesInternal(@TenantAwareParam String crn, String
         sdxBackupRestoreService.submitDatalakeDataInfo(operationId, dataSizesJson, ThreadBasedUserCrnProvider.getUserCrn());
     }
 
+    @Override
+    public FlowIdentifier diskUpdateByName(String name, DiskUpdateRequest updateRequest) {
+        String userCrn = ThreadBasedUserCrnProvider.getUserCrn();
+        SdxCluster sdxCluster = getSdxClusterByName(name);
+        updateRequest.setStackId(sdxCluster.getStackId());
```

Comment:
why do we need stackId in the request if we change it here?

Reply:
The request when received will not have stack id - i am setting the stack id and then passing it on to the Flow.

Reply:
then add it as a method parameter. manipulating request from service is an antipattern.

Reply:
and you can remove stackid from request if it is empty